src/org/mozilla/javascript/NativeArray.java

```diff
@@ -151,6 +151,7 @@ protected void fillConstructorProperties(IdFunctionObject ctor) {
         addIdFunctionProperty(ctor, ARRAY_TAG, ConstructorId_findIndex, "findIndex", 1);
         addIdFunctionProperty(ctor, ARRAY_TAG, ConstructorId_reduce, "reduce", 1);
         addIdFunctionProperty(ctor, ARRAY_TAG, ConstructorId_reduceRight, "reduceRight", 1);
+        addIdFunctionProperty(ctor, ARRAY_TAG, ConstructorId_flat, "flat", 0);
```

Comment:
This code adds the method to the constructor(`Array.flat`), not the prototype(`Array.prototype.flat`).
It is an old specification called Array generic methods.
https://lia.disi.unibo.it/materiale/JS/developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array.html#Array_generic_methods

I do not think this is necessary as it is not in latest ECMA262 specification.

Reply:
Right, I just added these because I saw other functions doing this -- e.g. `Array.reduceRight` doesn't exist either, just `Array.prototype.reduceRight`.

I've removed `ConstructorId_flat` entirely.